### PR TITLE
Added ability to wait before triggering a downstream build if current status is "running"

### PR DESCRIPTION
**Pull request by bchivari**
I'm not sure if you'd like to include this upstream but it has proved quite useful for me when dealing with downstream projects that have more than one upstream dependency that could trigger a build.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"fmt"
 	"os"
-
+	"time"
```
Comment (tboerger) [inline main.go]: Please keep the empty line between stdlib and other deps

File: main.go [hunk 2]
```diff
@@ -38,6 +38,17 @@ func main() {
...
 			EnvVar: "PLUGIN_FORK",
 		},
+		cli.BoolFlag{
+			Name:   "wait",
+			Usage:  "Wait for any currently running builds to finish",
+			EnvVar: "PLUGIN_WAIT",
+		},
+		cli.DurationFlag{
+			Name:   "timeout",
+			Value:  time.Duration(60)*time.Second,
```
Comment (tboerger) [inline main.go]: Have you go fmt the code?

File: main.go [hunk 3]
```diff
@@ -59,6 +70,8 @@ func run(c *cli.Context) error {
 		Server: c.String("server"),
 		Token:  c.String("token"),
 		Fork:   c.Bool("fork"),
+		Wait:   c.Bool("wait"),
+		Timeout:   c.Duration("timeout"),
```
Comment (tboerger) [inline main.go]: Missing go fmt

File: plugin.go [hunk 4]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"fmt"
 	"strings"
-
+	"time"
```
Comment (tboerger) [inline plugin.go]: Same as above

File: plugin.go [hunk 5]
```diff
@@ -13,11 +13,12 @@ type Plugin struct {
 	Server string
 	Token  string
 	Fork   bool
+	Wait   bool
+	Timeout  time.Duration
```
Comment (tboerger) [inline plugin.go]: Same as above

File: plugin.go [hunk 6]
```diff
@@ -29,46 +30,56 @@ func (p *Plugin) Exec() error {
...
+		tick := time.Tick(500 * time.Millisecond)
+	
+		// Keep trying until we're timed out, successful or got an error
+		// Tagged with "I" due to break nested in select
+                I:
+		for {
+			select {
+			// Got a timeout! fail with a timeout error
+			case <-timeout:
+				return fmt.Errorf("Error: timed out waiting on a build for %s.\n", entry)
```
Comment (tboerger) [inline plugin.go]: Waiting for? The unit?

File: plugin.go [hunk 7]
```diff
@@ -29,46 +30,56 @@ func (p *Plugin) Exec() error {
...
+						// rebuild the latest build
+						_, err = client.BuildStart(owner, name, build.Number)
+						if err != nil {
+							return fmt.Errorf("Error: unable to trigger build for %s.\n", entry)
+						}
+						fmt.Printf("Restarting build %d for %s\n", build.Number, entry)
+                                                break I;
+					}
+				} else if p.Wait == true {
+					fmt.Printf("Waiting on running build for: " + entry + "\n")
```
Comment (tboerger) [inline plugin.go]: Waiting for? The unit?

File: plugin.go (hunk 6 above)
Comment (bchivari) [inline plugin.go]: I've added some clarification to the output indicating what is being waited on. Just let me know if it remains unclear.

**Comment by bchivari** [comment]
Updated as per review from @tboerger

File: plugin.go (hunk 6 above)
Comment (tboerger) [inline plugin.go]: Maybe write seconds at the end?

File: plugin.go [hunk 8]
```diff
@@ -35,37 +37,49 @@ func (p *Plugin) Exec() error {
...
+						// rebuild the latest build
+						_, err = client.BuildStart(owner, name, build.Number)
+						if err != nil {
+							return fmt.Errorf("Error: unable to trigger build for %s.\n", entry)
+						}
+						fmt.Printf("Restarting build %d for %s\n", build.Number, entry)
+						break I
+					}
+				} else if p.Wait == true {
+					fmt.Printf("BuildLast for repository: %s, returned build number: %v with a status of %s. Will retry for %v.\n", entry, build.Number, build.Status, p.Timeout)
```
Comment (tboerger) [inline plugin.go]: Seconds

Comment (bchivari) [inline plugin.go]: Printing the timeout does display the unit (s,m,h...) by default.

**Comment by tboerger** [comment]
LGTM

**Comment by bchivari** [comment]
Any updates on this?

**Comment by tboerger** [comment]
@bradrydzewski ping

**Comment by tboerger** [comment]
@drone-plugins/maintainers can somebody of you also take a look at this?

**Comment by delfer** [comment]
LGTM

**Comment by vtolstov** [comment]
Nice to have this feature. With this plugin i'm able to get serial builds

File: plugin.go [hunk 9]
```diff
@@ -35,37 +37,49 @@ func (p *Plugin) Exec() error {
...
+						}
+						fmt.Printf("Starting new build %d for %s.\n", build.Number, entry)
+						break I
+					} else {
+						// rebuild the latest build
+						_, err = client.BuildStart(owner, name, build.Number)
+						if err != nil {
+							return fmt.Errorf("Error: unable to trigger build for %s.\n", entry)
+						}
+						fmt.Printf("Restarting build %d for %s\n", build.Number, entry)
```
Comment (MOZGIII) [inline plugin.go]: Bugged error reporting

**Pull request merged**